packages/@lwc/template-compiler/src/parser-next/index.ts

```diff
@@ -0,0 +1,928 @@
+        }
+    }
+
+    element.slotName = name;
+
+    const alreadySeen = ctx.seenSlots.has(name);
+    ctx.seenSlots.add(name);
+
+    if (alreadySeen) {
+        return ctx.warnOnIRNode(ParserDiagnostics.NO_DUPLICATE_SLOTS, element, [
+            name === '' ? 'default' : `name="${name}"`,
+        ]);
+    } else if (isInIteration(ctx, element)) {
+        return ctx.warnOnIRNode(ParserDiagnostics.NO_SLOTS_IN_ITERATOR, element, [
+            name === '' ? 'default' : `name="${name}"`,
+        ]);
+    }
+}
+
+function applyAttributes(ctx: ParserCtx, element: IRElement, parsedAttr: ParsedAttribute) {
+    const { tag } = element;
+    const attributes = parsedAttr.getAttributes();
+
+    for (const attr of attributes) {
+        const { name } = attr;
+
+        if (!isValidHTMLAttribute(tag, name)) {
+            ctx.warnOnIRNode(ParserDiagnostics.INVALID_HTML_ATTRIBUTE, attr, [name, tag]);
+        }
+
+        if (name.match(/[^a-z0-9]$/)) {
+            ctx.throwOnIRNode(
+                ParserDiagnostics.ATTRIBUTE_NAME_MUST_END_WITH_ALPHA_NUMERIC_CHARACTER,
+                attr,
+                [name, tag]
+            );
+        }
+
+        if (!/^-*[a-z]/.test(name)) {
+            ctx.throwOnIRNode(
+                ParserDiagnostics.ATTRIBUTE_NAME_MUST_START_WITH_ALPHABETIC_OR_HYPHEN_CHARACTER,
+                attr,
+                [name, tag]
+            );
+        }
+
+        // disallow attr name which combines underscore character with special character.
+        // We normalize camel-cased names with underscores caMel -> ca-mel; thus sanitization.
+        if (name.match(/_[^a-z0-9]|[^a-z0-9]_/)) {
+            ctx.throwOnIRNode(
+                ParserDiagnostics.ATTRIBUTE_NAME_CANNOT_COMBINE_UNDERSCORE_WITH_SPECIAL_CHARS,
+                attr,
+                [name, tag]
+            );
+        }
+
+        if (isIRStringAttribute(attr)) {
+            if (name === 'id') {
+                const { value } = attr;
+
+                if (/\s+/.test(value)) {
+                    ctx.throwOnIRNode(ParserDiagnostics.INVALID_ID_ATTRIBUTE, attr, [value]);
+                }
+
+                if (isInIteration(ctx, element)) {
+                    ctx.throwOnIRNode(ParserDiagnostics.INVALID_STATIC_ID_IN_ITERATION, attr, [
+                        value,
+                    ]);
+                }
+
+                if (ctx.seenIds.has(value)) {
+                    ctx.throwOnIRNode(ParserDiagnostics.DUPLICATE_ID_FOUND, attr, [value]);
+                } else {
+                    ctx.seenIds.add(value);
+                }
+            }
+        }
+
+        // Prevent usage of the slot attribute with expression.
+        if (name === 'slot' && isIRExpressionAttribute(attr)) {
+            ctx.throwOnIRNode(ParserDiagnostics.SLOT_ATTRIBUTE_CANNOT_BE_EXPRESSION, attr);
+        }
+
+        // the if branch handles
+        // 1. All attributes for standard elements except 1 case are handled as attributes
```

Comment:
In the new shape of the AST the `Element` only has [Attributes](https://github.com/salesforce/lwc/blob/ce3fb5b8822ac5dbe71e6165ea7b65f4fbf54c67/packages/%40lwc/template-compiler/src/shared-next/types.ts#L148).  However, it looks like there's[ one case ](https://github.com/salesforce/lwc/blob/ce3fb5b8822ac5dbe71e6165ea7b65f4fbf54c67/packages/%40lwc/template-compiler/src/parser-next/attribute.ts#L209)where the `Element` should have a `Property`.   I'm considering moving the single case to just be an `Attribute` but I'm not sure if there's any downstream effects to this.

Do you think I should keep the single case as a `Property` or is it ok to move to an `Attribute`?